### PR TITLE
jobs: fix job 'running state'

### PR DESCRIPTION
**Pull request by Exirel**
### Description

For some reason it was broken? I don't understand what happened here. I'll probably have to investigate further to understand the "why did I do that?" later. That will do at the moment, so no worries.

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

**Comment by dgw** [comment]
@Exirel I came back to properly review this but the docstring fix isn't done yet. :S

Guess the real question is: Is a threaded job _always_ considered `is_running` even if it's, um, idle? Not sure I understand the intended behavior or use of the flag, then. 😕

**Comment by Exirel** [comment]
@dgw the goal is to prevent such job to run multiple time in parallel. Like, you have a job and you want it in a thread because it takes, like, 30s. It's fine, since it' should run only every 5min, but then you have a very weird case where it takes more than 5 mins, and "oh no" you don't want it to run again.

I think I wanted to be "too smart" with the `with job: job.execute()` thing, and it didn't come out as intended initially.

In short: sloppy job on my part. Will fix the docstring asap. Soon. This week.
